### ISSUE TITLE
Version 2.03 does not compile properly on Windows

### ISSUE THREAD

**Issue by GoogleCodeExporter** [sbmbeng/tesseract-ocr#198]
```
What steps will reproduce the problem?
1. Download tesseract-2.03, run configure, then try to run make.  I've 
done this on both a Windows XP Pro 32-bit system, and a Windows Vista 64-
bit system with the same bad result.  Both machines are using the most up-
to-date version of cygwin.

What is the expected output? What do you see instead?
Compilation works fine until you reach the viewer directory.  Compiling 
svutil.cpp fails with the following complaint:
{{{
svutil.cpp: In constructor `SVNetwork::SVNetwork(const char*, int)':
svutil.cpp:245: error: aggregate `sockaddr_in address' has incomplete type 
and cannot be defined
svutil.cpp:273: error: `htons' undeclared (first use this function)
svutil.cpp:273: error: (Each undeclared identifier is reported only once 
for each function it appears in.)
}}}

Tracing through the code it appears that the problem is that the symbol 
WIN32 is not defined.  The compiler command is:
{{{
g++ -DHAVE_CONFIG_H -I. -I. -I.. -g -O2 svutil.cpp
}}}

The Visual Studio project does build this file successfully - because it 
defines WIN32.  However, note that simply defining WIN32 on the command 
line for g++ does not directly fix the problem.  For example, the command:
{{{
g++ -DHAVE_CONFIG_H -DWIN32 -I. -I. -I.. -g -O2 svutil.cpp
}}}
fails for a new reason:
{{{
In file included from svutil.cpp:27:
/usr/lib/gcc/i686-pc-
cygwin/3.4.4/../../../../include/w32api/winsock.h:82:2: warning: 
#warning "fd_set and associated macros have been defined in 
sys/types.      This can cause runtime problems with W32 sockets"
In file included from svutil.cpp:27:
/usr/lib/gcc/i686-pc-
cygwin/3.4.4/../../../../include/w32api/winsock.h:480: error: declaration 
of C function `int gethostname(char*, int)' conflicts with
/usr/include/sys/unistd.h:206: error: previous declaration `int gethostname
(char*, size_t)' here
}}}
I'm not sure I understand the first error message, but the second one is 
due to the fact that on Windows size_t is unsigned.  The Visual Studio 
project compiles successfully because it includes the Microsoft versions 
of <windows.h> and <winsock.h>, which are different than the GNU versions 
of the same include files.

What version of the product are you using? On what operating system?
Tesseract 2.03 with the latest cygwin (full installation) and gcc 3.4.4, 
with either Windows XP Pro SP3 32-bit, and with Windows Vista 64-bit.

Please provide any additional information below.
Note that I've filed a separate issue for the Visual Studio version (only 
minor issues).
```

Original issue reported on code.google.com by `namefun...@gmail.com` on 31 Mar 2009 at 4:32
